custom_components/ha_kia_hyundai/services.py

```diff
@@ -24,23 +30,40 @@
 
 _LOGGER = getLogger(__name__)
 
+
 def async_setup_services(hass: HomeAssistant):
     async def async_handle_start_climate(call: ServiceCall):
         coordinator: VehicleCoordinator = _get_coordinator_from_device(hass, call)
         climate = call.data.get(SERVICE_ATTRIBUTE_CLIMATE)
         set_temp = call.data.get(SERVICE_ATTRIBUTE_TEMPERATURE)
         defrost = call.data.get(SERVICE_ATTRIBUTE_DEFROST)
         heating = call.data.get(SERVICE_ATTRIBUTE_HEATING)
+        driver_seat = call.data.get(SERVICE_ATTRIBUTE_DRIVER_SEAT, None)
+        passenger_seat = call.data.get(SERVICE_ATTRIBUTE_PASSENGER_SEAT, None)
+        left_rear_seat = call.data.get(SERVICE_ATTRIBUTE_LEFT_REAR_SEAT, None)
+        right_rear_seat = call.data.get(SERVICE_ATTRIBUTE_RIGHT_REAR_SEAT, None)
 
         if set_temp is not None:
             set_temp = int(set_temp)
+        if driver_seat is not None:
+            driver_seat = SeatSettings(driver_seat)
```

Comment:
guess the renumbering would still reck havoc here anyway so don't worry about the other comment either way is fine

Reply:
might be a good argument for the service accepting strings instead of numbers though.  As this isn't a feature I expect to use though I'm okay either way, thanks for all your improvements to this repo

Reply:
looks like the hass validation failure is also related, though you can probably pass that with just quotes around the number in the yaml, but switching it to strings would also make the hass validator happy.